### DISCUSSION TITLE
How about adding temporary window rule for starting application?

### DISCUSSION THREAD

**Discussion by Ziqi-Yang** [hyprwm/Hyprland#1395] [General]
# scene
1. Start applications at Hyprland logging and set them to custom workspace and more.
2. More flexibility and make the configuration file more concise. For example, you want to start an application fullscreen at any workspace, but you already have a window rule in the configuration file that makes this application float in the second workspace. If the application doesn't provide starting window class option and cannot change the start window title, I think it can be difficult to do that work.

# description
The window rule can be temporarily applied to starting an application by using `hyprctl`(adding a new command like `exec` but combined with window rule syntax? ). The window rule should be universal to all applications that will be started in the command(or just the window belongs to the parent process).

**Comment by vaxerski** [comment]
uuuhhhh

`bind = SUPER, A, exec, [workspace 2] kitty`

`hyprctl dispatch exec "[workspace 2] kitty"`

`exec-once = [workspace 2] kitty`

????

**Reply by MightyPlaza** [reply]
```
hyprctl dispatch exec "[workspace 2\; float] kitty"
exec-once = [workspace 2; float] kitty
```

**Reply by contre95** [reply]
Oh sorry, I deleted comment. Just found it in the docs https://wiki.hyprland.org/Configuring/Dispatchers/#executing-with-rules. Thanks @vaxerski 
For context: My question was about how to add more rules between the `[]`.

**Comment by Ziqi-Yang** [comment]
[choke] x⸑x
